sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -306,6 +306,7 @@ case class FileSourceScanExec(
   }
 
   private lazy val inputRDD: RDD[InternalRow] = {
+    DataSourceUtils.verifyReadSchema(relation.fileFormat, relation.dataSchema)
```

Comment:
In some formats, is this verification applied two times, right?

Reply:
I'm not sure about this change. This is very late(just before execution), is there a better place for this check that happens at analysis phase?

Reply:
BTW why do we need to check schema at read path?

Reply:
Yes, for user-specified schema.